### PR TITLE
(Fixes #447) Fix the conflict of ami and release label

### LINKED ISSUE #447: EMR cluster when using configuration to specify release label, the ami field is not overriden
It should shadow the AMI configuration if the release label field is present

### PR DESCRIPTION
**Pull request by realstraw**
- when both exists in configs

### CODE REVIEW

File: core/src/main/scala/com/krux/hyperion/resource/EmrCluster.scala [hunk 1]
```diff
@@ -15,7 +15,7 @@ trait EmrCluster extends ResourceObject {
   def emrClusterFields: EmrClusterFields
   def updateEmrClusterFields(fields: EmrClusterFields): Self
 
-  def amiVersion = emrClusterFields.amiVersion
+  def amiVersion = if (releaseLabel.nonEmpty) None else emrClusterFields.amiVersion
```
Comment (sethyates) [inline core/src/main/scala/com/krux/hyperion/resource/EmrCluster.scala]: How is this happening to begin with?  Below, `withAmiVersion` resets `releaseLabel` to `None`.  Is it actually a bug in the `apply`?

Comment (realstraw) [inline core/src/main/scala/com/krux/hyperion/resource/EmrCluster.scala]: when config file have both ami and release label specified.

Comment (realstraw) [inline core/src/main/scala/com/krux/hyperion/resource/EmrCluster.scala]: yes, it's because of `apply` should we fix it there?

Comment (realstraw) [inline core/src/main/scala/com/krux/hyperion/resource/EmrCluster.scala]: actually, not `apply` but `defaultEmrClusterFields`

Comment (sethyates) [inline core/src/main/scala/com/krux/hyperion/resource/EmrCluster.scala]: Yeah I'd fix it in `defaultEmrClusterFields`.

Comment (realstraw) [inline core/src/main/scala/com/krux/hyperion/resource/EmrCluster.scala]: Done

**Comment by jnewman** [comment]
LGTM

**Comment by sethyates** [comment]
LGTM

**Pull request merged**